File: legacy/legacy-mahabharata-android/app/src/main/java/com/fulldome/mahabharata/model/visual/animation/AlphaAnim.java
package com.fulldome.mahabharata.model.visual.animation;

import androidx.annotation.FloatRange;

import com.fulldome.mahabharata.model.visual.Layer;

public class AlphaAnim extends LayerAnim {
	private float alpha;

	public AlphaAnim() {
	}

	public AlphaAnim(@FloatRange(from = 0f, to = 1f) float alpha) {
		this.alpha = alpha;
	}

	@FloatRange(from = 0f, to = 1f)
	public float getAlpha() {
		return alpha;
	}

	@Override
	protected AlphaAnim interpolate(LayerAnim endAnim, float fraction) {
		AlphaAnim end = (AlphaAnim) endAnim;
		return new AlphaAnim(Layer.FLOAT_EVALUATOR.evaluate(fraction, getAlpha(), end.getAlpha()));
	}

	@Override
	public void apply(Layer.ViewData data, int width, int height) {
		data.setAlpha(alpha);
	}
}

File: legacy/legacy-bhagavadgita.comics-ios-swift/Mahabharata/Model/DataClasses/Visual/Animations/AlphaAnim.swift
//
//  Comics.swift
//  Mahabharata
//
//  Created by Roman Developer on 11/28/17.
//  Copyright © 2017 Iron Water Studio. All rights reserved.
//

import UIKit

class AlphaAnim: Anim {
	let alpha: Double
	
	private enum CodingKeys: String, CodingKey {
		case alpha
	}
	
	required init(from decoder: Decoder) throws {
		let container = try decoder.container(keyedBy: CodingKeys.self)
		self.alpha = (try? container.decode(Double.self, forKey: .alpha)) ?? 0

		try super.init(from: decoder)
	}
	
	init(alpha: Double) {
		self.alpha = alpha
		
		super.init()
	}
	
	override var description: String {
		return [
			super.description,
			"Alpha: \(alpha)"
			].joined(separator: ", ")
	}
	
	//MARK: - Overrides
	override func interpolate(endAnim: Anim, fraction: Double) -> Anim {
		let end = endAnim as! AlphaAnim
		return AlphaAnim(alpha: LinearInterpolation.evaluate(fraction: fraction, startValue: self.alpha, endValue: end.alpha))
	}
	
	override func apply(to data: Layer.ViewData, width: Int, height: Int) {
		data.alpha = self.alpha
	}
}

